typed-racket-test/succeed/gh-issue-930.rkt

```diff
@@ -0,0 +1,13 @@
+#lang typed/racket/base
```

Comment:
Need tests for all the things that worked previously (ie, using a type name a struct name, which was a bug).